src/structs.py

```diff
@@ -222,10 +222,16 @@ def variables(self):
         return list(self.entities)
 
     @cached_property
-    def _str(self):
+    def _str(self) -> str:
         return (str(self.predicate) + "(" +
                 ", ".join(map(str, self.variables)) + ")")
 
+    def ground(self, sub) -> GroundAtom:
```

Comment:
Would we ever need to go the other way, i.e. GroundAtom to LiftedAtom? We would need this sometimes right, like when we lift effects for clustering?